forum/projects/views.py

```diff
@@ -1,3 +1,16 @@
-from django.shortcuts import render
+from rest_framework.viewsets import ModelViewSet
+from rest_framework.permissions import IsAuthenticated
+from .models import Project
+from .serializers import ProjectSerializer
+from .permissions import IsOwnerOrReadOnly
 
-# Create your views here.
+
+class ProjectViewSet(ModelViewSet):
+    permission_classes = [IsAuthenticated, IsOwnerOrReadOnly]
+    queryset = Project.objects.all()
+    serializer_class = ProjectSerializer
+
+    def perform_create(self, serializer):
+        """Automatically assigns project to correct startup based on user's token"""
+        startup = self.request.user.startup_profile
+        serializer.save(startup=startup)
```

Comment:
Add error handling to ensure the user has a startup profile. Without this, a missing profile will raise an unhandled exception.